src/wp-includes/blocks.php

```diff
@@ -2083,3 +2083,98 @@ function _wp_footnotes_force_filtered_html_on_import_filter( $arg ) {
 	}
 	return $arg;
 }
+
+
+/**
+ * Processes the block bindings in block's attributes.
+ *
+ * A block might contain bindings in its attributes. Bindings are mappings
+ * between an attribute of the block and a source. A "source" is a function
+ * registered with `wp_block_bindings_register_source()` that defines how to
+ * retrieve a value from outside the block, e.g. from post meta.
+ *
+ * This function will process those bindings and replace the HTML with the value of the binding.
+ * The value is retrieved from the source of the binding.
+ *
+ * ### Example
+ *
+ * The "bindings" property for an Image block might look like this:
+ *
+ * ```json
+ * {
+ *   "metadata": {
+ *     "bindings": {
+ *       "title": {
+ *         "source": {
+ *           "name": "post_meta",
+ *           "attributes": { "value": "text_custom_field" }
+ *         }
+ *       },
+ *       "url": {
+ *         "source": {
+ *           "name": "post_meta",
+ *           "attributes": { "value": "url_custom_field" }
+ *         }
+ *       }
+ *     }
+ *   }
+ * }
+ * ```
+ *
+ * The above example will replace the `title` and `url` attributes of the Image
+ * block with the values of the `text_custom_field` and `url_custom_field` post meta.
+ *
+ * @access private
+ * @since 6.5.0
+ *
+ * @param string   $block_content Block content.
+ * @param array    $block The full block, including name and attributes.
+ * @param WP_Block $block_instance The block instance.
+ */
+function _process_block_bindings( $block_content, $block, $block_instance ) {
```

Comment:
This method should be moved to the `WP_Block` class as a private method. It's used only there, and we don't want to keep it as a public API.

As the next step that can be done seperately in the future, we should also consider splitting this method into two steps:
- updating block attributes on the `WP_Block` instance somewhere here:
  https://github.com/WordPress/wordpress-develop/blob/2e3b292924e98fd8c4e94bd96f3e0ac6ca54d198/src/wp-includes/class-wp-block.php#L179-L189
- replacing the bindings using the computed attributes in a similar place as in this PR